plugins/etcd/src/main/java/org/radargun/service/EtcdService.java

```diff
@@ -17,17 +25,59 @@
  * Note that etcd saves some data in the distribution directory, and you need to delete it between runs. Using the distributionZip
  * property is preferred, so the distribution is 'clean' with each run.
  *
+ * The documentation about the the configuration is available at https://coreos.com/etcd/docs/latest/op-guide/configuration.html
+ *
  */
 @Service(doc = "Etcd")
 public class EtcdService implements Lifecycle {
 
    private Process server;
 
-   @Property(doc = "Directory of the etcd distribution.", optional = false)
+   @Property(doc = "Distribution zip. If set, the zip will be extracted to the 'distributionDir' directory.", optional = false)
+   protected String distributionZip;
+
+   /**
+    * It is realted with the --discovery member flag.
+    */
+   @Property(doc = "Prefix flags need to be set when using discovery service.", optional = false)
+   protected String discovery;
```

Comment:
We are using public discovery service https://github.com/coreos/etcd/blob/master/Documentation/op-guide/clustering.md#public-etcd-discovery-service

In this case, we need to execute the following script before start the test
ETCD_DISCOVERY_URL=$(curl https://discovery.etcd.io/new?size=3)

In the RadarGun config file we need to set: discovery="${env.ETCD_DISCOVERY_URL}"

Reply:
Oh. This looks like something that could be done automatically.

Reply:
But, to be honest. I don't insist on that.

Reply:
The public discovery service is nice for demos, but could we start the discovery service directly in the cluster?

Reply:
Per my understanding we have 3 options:
Static - Need to know all the clusters members before start each node
etcd Discovery - We can create a private or use the public
DNS Discovery - We need to create DNS SRV records

We can built our private etcd discovery using the source code: https://github.com/coreos/discovery.etcd.io

IMO, I will prefer use the public discovery service and if for some reason it goes down or stop working we can create our server.

I was not able to find an way where we can start a discovery service directly in the cluster.

Reply:
OK, if it's not easy to set it up yourselves, let's forget about it.

Note that rather than setting env vars manually (you need call the curl on single node and pass that to all slaves, don't you?) you can run the command from benchmark itself using `<command cmd="curl" args="https://discovery.etcd.io/new?size=3" var="discovery.url"/>`. That would run it from all slaves, though; you could enhance that stage to distribute the result to all nodes overriding `processAckOnMaster`. That should go to different PR, though.

Reply:
Sounds good to me. It will fix the @mgencur suggestion as well.

https://github.com/radargun/radargun/issues/531